File: leet_1614_max_depth_parentheses.cpp
class Solution {
public:
    int maxDepth(string s)
    {
        int r = 0, t = 0;

        for (char c : s)
        {
            if (c == '(')
                ++t;
            if (c == ')')
                --t;
            if (r < t)
                r = t;
        }
        return r;
    }
};

File: leet_1614_max_depth_parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        r, t = 0, 0
        for c in s:
            if c is '(':
                t += 1
            if c is ')':
                t -= 1
            if t > r:
                r = t
        return r
